azure-pipelines/main-pipeline.yml

```diff
@@ -0,0 +1,228 @@
+# The idea is to be inclusive of all source directories. (I.E. Commerce, Accounts, Eureka, Gateway, Inventory, Config, and Base(?))
+# And using the variable to set a condition that will set pipeline variables variables that will control the flow of the pipeline.
+
+trigger:
+
+  branches:
+    include: 
+    - experimental-pipeline
+
+pr:
+  branches:
+    include:
+    - main
+    - staging
+    - development
+    - feature_*
+    
+
+# Parameters and Variables to help with naming and calling variables, as well as providing a group of objects to loop over in later jobs.
+variables:
+    
+    org: '1054-daniel-felleman-revature-swag-shop'
+    projectKey: '1054-Daniel-Felleman-Revature-Swag-Shop_RevatureShopBackend'
+    projectName: 'RevatureShopBackend'
+
+parameters:
+- name: serviceConditions
+  type: object
+  default:
+  - accounts:
+    name: 'accounts'
+    requiresBase: true
+    jobName: 'AccountsBuild'
+  - commerce:
+    name: 'commerce'
+    requiresBase: true
+    jobName: 'CommerceBuild'
+  - config:
+    name: 'config'
+    jobName: 'ConfigBuild'
+  - eureka-server:
+    name: 'eureka-server'
+    jobName: 'EurekaBuild'
+  - inventory:
+    name: 'inventory'
+    requiresBase: true
+    jobName: 'InventoryBuild'
+  - Spring-Cloud-Gateway:
+    name: 'Spring-Cloud-Gateway'
+    jobName: 'SpringBuild'
+
+stages:
+- stage: Check
+  displayName: Check Service Changes, Test and Build
+  jobs:
+  - job: CheckChanges
+    pool: java-16
+    steps:
+
+# This script will check the Repo for changes and store them in a variable updatedFiles.
+# The script will iterate over every object stored in updatedFiles to select paths with changes.
+# Pipeline variable will be updated from 'false' to 'True' if a change was made in that path.
+# This step controls which build tasks will be triggered.
+# If Base is to be rebuilt, it will then set those services for which it is a dependency to build as well.   
+
+    - powershell: |
+        $updatedFiles = git diff HEAD HEAD~ --name-only
```

Comment:
there are possible edge-cases here based on [this behavior](https://git-scm.com/docs/git-config#Documentation/git-config.txt-corequotePath) which could be addressed by adding the `-z` option and making sure it's splitting on null bytes and not on newlines. or by ensuring the below logic can handle when the filenames are quoted and escaped.

Reply:
Checking this out!

Reply:
-z, according to the docs works only with --options I'm not using. I will continue to research this.